monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringDamGenerator/MonitoringDamGenerator.java

```diff
@@ -0,0 +1,249 @@
+package eu.seaclouds.monitor.monitoringDamGenerator;
+
+import it.polimi.tower4clouds.rules.MonitoringRules;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import eu.seaclouds.monitor.monitoringDamGenerator.adpParsing.Host;
+import eu.seaclouds.monitor.monitoringDamGenerator.adpParsing.Module;
+import eu.seaclouds.monitor.monitoringDamGenerator.adpParsing.ParsingException;
+import eu.seaclouds.monitor.monitoringDamGenerator.adpParsing.YAMLMonitorParser;
+import eu.seaclouds.monitor.monitoringDamGenerator.dcScriptGenerators.JavaAppLevelDcDeploymentScriptGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.dcScriptGenerators.MODACloudsDcDeploymentScriptGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.dcScriptGenerators.NuroDcDeploymentScriptGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.rulesGenerators.ApplicationRulesGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.rulesGenerators.InfrastructuralRulesGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.rulesGenerators.NuroRulesGenerator;
+import eu.seaclouds.monitor.monitoringDamGenerator.util.CustomizedApplications;
+
+public class MonitoringDamGenerator {
+    private static final Logger logger = LoggerFactory
+            .getLogger(MonitoringDamGenerator.class);
+
+    private static String monitoringManagerIp;
+    private static String monitoringManagerPort;
+    
+    private static final String NURO_DC_ID="nuroDc";
+    private static final String JAVA_APP_DC_ID="javaAppDc";
+    private static final String MODACLOUDS_DC_ID="modacloudsDc";
+
+    public MonitoringDamGenerator(String monitorUrl, String monitorPort) {
+        
+        monitoringManagerIp = monitorUrl;
+        monitoringManagerPort = monitorPort;
+
+    }
+
+    public List<Module> generateMonitoringInfo(String adp) {
```

Comment:
Please add some javadoc as this method is doing much more than what the name declares :)